File: python/kode program/produk.py
class Produk:
     # Konstruktor dipanggil saat objek baru dibuat
     # Parameter default id=0, nama="", merk="", harga=0, stok=0
    def __init__(self, i=0, n="", m="", h=0, s=0): 
        self.__id = i #atribut private id produk
        self.__nama = n #atribut private nama produk
        self.__merk = m #atribut private merk produk
        self.__harga = h #atribut private harga produk
        self.__stok = s #atribut private stok produk

     # Setter 
    def setId(self, i): self.__id = i# ubah id produk
    def setNama(self, n): self.__nama = n# ubah nama produk
    def setMerk(self, m): self.__merk = m# ubah merk produk
    def setHarga(self, h): self.__harga = h# ubah harga produk
    def setStok(self, s): self.__stok = s# ubah stok produk

    # Getter 
    def getId(self): return self.__id # ambil id produk
    def getNama(self): return self.__nama# ambil nama produk
    def getMerk(self): return self.__merk# ambil merk produk
    def getHarga(self): return self.__harga# ambil harga produk
    def getStok(self): return self.__stok# ambil stok produk

    # menampilkan data
    def tampilkan(self):
        print(f"ID-> {self.__id} | Nama-> {self.__nama} | Merk-> {self.__merk} | Harga-> Rp{self.__harga} | Stok-> {self.__stok}")

File: php/kode program/produk.php
<?php
class Produk {
    // Deklarasi atribut bersifat private agar hanya bisa diakses lewat method getter & setter
    private $id; // menyimpan ID produk
    private $nama; // menyimpan nama produk
    private $merk; //menyimpan nama merk
    private $harga;// menyimpan harga produk
    private $stok;// menyimpan jumlah stok produk
    private $gambar; // menyimpan path/nama file gambar produk

     // Constructor otomatis dipanggil ketika objek Produk dibuat
    public function __construct($id, $nama, $merk, $harga, $stok, $gambar) {
        $this->id = $id; // inisialisasi ID
        $this->nama = $nama; //inisialisasi nama
        $this->merk = $merk; //inisialisasi merk
        $this->harga = (float)$harga; // inisialisasi harga
        $this->stok = $stok; // inisialisasi stok
        $this->gambar = $gambar; // inisialisasi gambar
    }

    // getter & Setter
    //mengambil nilai id produk
    public function getId() {
        return $this->id;
    }
    //mengubah nilai id produk
    public function setId($id) {
        $this->id = $id;
    }

    //mengambil nilai nama produk
    public function getNama() {
        return $this->nama;
    }
    //mengubah nilai nama produk
    public function setNama($nama) {
        $this->nama = $nama;
    }

    //mengambil nilai merk produk
    public function getMerk() {
        return $this->merk;
    }
    //mengubah nilai merk produk
    public function setMerk($merk) {
        $this->merk = $merk;
    }

    //mengambil nilai harga produk
    public function getHarga() {
        return $this->harga;
    }
    //mengubah nilai harga produk
    public function setHarga($harga) {
        $this->harga = (float)$harga;
    }

    //mengambil nilai stok produk
    public function getStok() {
        return $this->stok;
    }
    //mengubah nilai stok produk
    public function setStok($stok) {
        $this->stok = $stok;
    }

    //mengambil nilai gambar produk
    public function getGambar() {
        return $this->gambar;
    }
    //mengubah nilai gambar produk
    public function setGambar($gambar) {
        $this->gambar = $gambar;
    }
}
?>
